### ISSUE TITLE
`tests/tail-2/inotify-dir-recreate` fails intermittently

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#4294]
Full log: https://pipelines.actions.githubusercontent.com/serviceHosts/91055393-6e6f-4662-ba97-3acee74a8c7c/_apis/pipelines/1/runs/21407/signedlogcontent/2?urlExpires=2023-01-22T20%3A49%3A47.5591326Z&urlSigningMethod=HMACV1&urlSignature=3bdNcp%2FrkIaVp81kNK6%2F1nv7%2BLWMBOKkAYcJbxB5Fk4%3D

Fails with:

```
2023-01-07T09:27:55.8639630Z FAIL: tests/tail-2/inotify-dir-recreate
2023-01-07T09:27:55.8639861Z =======================================
2023-01-07T09:27:55.8639994Z 
2023-01-07T09:27:55.8640107Z tail did not switch to polling mode
2023-01-07T09:27:55.8640368Z previously removed file did not appear
2023-01-07T09:27:55.8640681Z --- exp	2023-01-07 09:03:09.501350606 +0000
2023-01-07T09:27:55.8640981Z +++ out	2023-01-07 09:03:03.197274646 +0000
2023-01-07T09:27:55.8641230Z @@ -1,5 +1,2 @@
2023-01-07T09:27:55.8641402Z  inotify
2023-01-07T09:27:55.8641745Z  tail: 'dir/file' has become inaccessible: No such file or directory
2023-01-07T09:27:55.8642133Z -tail: directory containing watched file was removed
2023-01-07T09:27:55.8642505Z -tail: inotify cannot be used, reverting to polling
2023-01-07T09:27:55.8642854Z -tail: 'dir/file' has appeared;  following new file
2023-01-07T09:27:55.8643245Z FAIL tests/tail-2/inotify-dir-recreate.sh (exit status: 1)
```

**Comment by Joining7943** [comment]
I spent some time searching for the reason of this and although I haven't yet found the cause for these intermittent failures, I'll share what I did: 

I tried to figure out which merge could have possibly caused these intermittent failure and
to narrow down the source of the failures, I thought I saw this intermittent failure the first time after #4201 (date of merge was `2022-11-29`) was merged into main. I searched through the ci logs backwards based on this (using the filter `created:<2022-11-29 workflow:GnuTests`) and the earliest failure I could find was https://github.com/uutils/coreutils/actions/runs/3492232792/jobs/5845780148#logs on Nov 17, 2022. This branch is still not merged into main and therefore can't be the root of the failure. So, I searched through the merged prs around that date (`is:pr merged:<2022-11-18`) and none of them were related to `tail`.

In the end, I haven't found out a definite reason for these intermittent failures being related to changes in `uutils`.
